### PR TITLE
Contract Versions

### PR DESCRIPTION
**Pull request by RafilxTenfen**
### Add version to contracts

- It was created a way to have multiple versions of the contracts
- Today, inside the bridge directory we have the contracts in the root of the contracts directory or even inside other directories, like: 

- This way we have like `_old` contracts to test the deploy to new contracts when upgrading 
- So, the idea is to make it possible to have multiple versions of the same contracts doing in a way that we don't need to use `_old_old.sol` but instead we could use like `BridgeV1.sol` and `BridgeV2.sol` each time we are going to deploy a new contract we could create a new version inside a contract path, the last contract (more updated one) would not have any versions at the end, so it would be `Bridge.sol`
- The last version of the contract would not have any Version at the end, so if we happen to have 3 Federation contracts it would be the first called V1 the second V2 and the last one (more updated contract) would be just `Federation.sol`

### Description

- Created directories for contracts
- Removed previous directory
- Add test for upgrading from the `federationV2` to `federation` 

### How to Test

- Run both `ganache-cli`
- Enter into the bridge directory and run all the tests

#### Case 1

1. Go to bridge directory
```shell
$~ cd bridge
```

2. Run the first ganache-cli
```shell
$~ npm run ganache
```

3. Open another shell and run the ganache mirror
```shell
$~ npm run ganache-mirror
```

4. Run all the tests
```shell
$~ npm run test
```
__Expected Result__
- It should pass the test
image

5. Deploy the contracts to the federator tests
```shell
$~ npm run deployLocalIntegrationTest
```

6. Go to the federator directory
```shell
$~ cd ../federator
```

7. Run the integration tests
```shell
$~ npm run integrationTest
```

#### Bridge Directory `cd bridge`
- [x] Lint is clean `npm run lint`
- [x] Test is passing `npm run test`
- [x] Contracts are compiling `npm run compile`

#### Federator Directory `cd federator`
- [x] Lint is clean `npm run lint`
- [x] Test is passing `npm run test`
- [x] Typescript is compiling `npm run build`

### CODE REVIEW

File: bridge/test/nftbridge/FederationV2_to_federation_test.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      let result = await proxyFederationV2.methods.version().call();
+      assert.equal(result, 'v2');
+
+      const federation = await Federation.new();
+      await proxyAdmin.upgrade(federationProxy.address, federation.address)
+      const proxyFederation = new web3.eth.Contract(federation.abi, federationProxy.address);
+      result = await proxyFederation.methods.version().call();
+      assert.equal(result, 'v3');
+
+      members = await proxyFederation.methods.getMembers().call();
```
Comment (pmprete) [inline bridge/test/nftbridge/FederationV2_to_federation_test.js:39]: you should add a test where it calls correctly to owner, primary and one to check if supports the new methods correctly after the upgrade

File: bridge/contracts/Bridge/BridgeV1.sol [hunk 2]
```diff
@@ -17,20 +17,20 @@ import "../zeppelin/token/ERC777/IERC777.sol";
...
+import "./IBridgeV1.sol";
 import "../interface/ISideToken.sol";
 import "../interface/ISideTokenFactory.sol";
-import "./AllowTokens_old.sol";
-import "./Utils_old.sol";
+import "../AllowTokens/AllowTokensV1.sol";
+import "../Utils/UtilsV1.sol";
 
-contract Bridge_old is Initializable, IBridge_old, IERC777Recipient, UpgradablePausable, UpgradableOwnable, ReentrancyGuard {
+contract BridgeV1 is Initializable, IBridgeV1, IERC777Recipient, UpgradablePausable, UpgradableOwnable, ReentrancyGuard {
```
Comment (pmprete) [inline bridge/contracts/Bridge/BridgeV1.sol:26]: This should be BridgeV2  https://github.com/rsksmart/tokenbridge/pull/263/files#diff-42699c4d471d9ab5f788196252f6ba98c1268374871e298566f1205f2ec1f655R73

**Review by pmprete — APPROVED** [review]
Except my comments loogs good 👍

File: bridge/test/nftbridge/FederationV2_to_federation_test.js (hunk 1 above)
Comment (RafilxTenfen) [reply bridge/test/nftbridge/FederationV2_to_federation_test.js:39]: Nice, will do

File: bridge/contracts/Bridge/BridgeV1.sol (hunk 2 above)
Comment (RafilxTenfen) [reply bridge/contracts/Bridge/BridgeV1.sol:26]: Should I add the Contract Bridge V1 as well? to the contract versions?

Comment (RafilxTenfen) [reply bridge/contracts/Bridge/BridgeV1.sol]: Done at https://github.com/rsksmart/tokenbridge/pull/263/commits/0dcd4497fb42c40e3cde1a253333b7ad68cc14ec

**Pull request merged**